### PR TITLE
CXF-8750: only add trailing slash if necessary when locating webjars …

### PR DESCRIPTION
**Pull request by awrb**
…resources for Swagger UI in OSGi environment

### CODE REVIEW

File: osgi/itests/src/test/java/org/apache/cxf/osgi/itests/jaxrs/JaxRsServiceTest.java [hunk 1]
```diff
@@ -99,6 +112,41 @@ public void testJaxRsPut() throws Exception {
...
+
+        try {
+            new ObjectMapper().readValue(openApiJson, Object.class);
+            assertTrue(openApiJson.contains("/bookstore/"));
+        } catch (JsonProcessingException e) {
+            fail();
+        }
+    }
+
+    private String readResponseAsString(Response response) {
```
Comment (reta) [inline osgi/itests/src/test/java/org/apache/cxf/osgi/itests/jaxrs/JaxRsServiceTest.java:143]: This is not necessary, you could just use `response.readEntity(String.class)` directly

File: osgi/itests/src/test/java/org/apache/cxf/osgi/itests/jaxrs/JaxRsServiceTest.java [hunk 2]
```diff
@@ -99,6 +112,41 @@ public void testJaxRsPut() throws Exception {
         assertStatus(Status.OK, response);
     }
 
+    @Test
+    public void testGetSwaggerUi() {
+        WebTarget swaggerWt = ClientBuilder.newClient().target(SWAGGER_URL)
+                .queryParam("url", "/cxf/jaxrs/openapi.json");
+        Response response = swaggerWt.request().get();
+        String swaggerFileHtml = readResponseAsString(response);
```
Comment (reta) [inline osgi/itests/src/test/java/org/apache/cxf/osgi/itests/jaxrs/JaxRsServiceTest.java:120]:
```suggestion
        String swaggerFileHtml = response.readEntity(String.class);
```

File: osgi/itests/src/test/java/org/apache/cxf/osgi/itests/jaxrs/JaxRsServiceTest.java [hunk 3]
```diff
@@ -99,6 +112,41 @@ public void testJaxRsPut() throws Exception {
...
+        assertTrue(swaggerFileHtml.contains("<head>"));
+        assertTrue(swaggerFileHtml.contains("<title>Swagger UI</title>"));
+        assertTrue(swaggerFileHtml.contains("</html>"));
+    }
+
+    @Test
+    public void testGetOpenApiJsonFile() {
+        WebTarget openApiWt = ClientBuilder.newClient().target(OPEN_API_FILE_URL);
+        Response response = openApiWt.request().get();
+        String openApiJson = readResponseAsString(response);
```
Comment (reta) [inline osgi/itests/src/test/java/org/apache/cxf/osgi/itests/jaxrs/JaxRsServiceTest.java:133]:
```suggestion
        String openApiJson = response.readEntity(String.class);
```

**Comment by reta** [comment]
Thanks @awrb , minor comments, LGTM otherwise!

**Comment by awrb** [comment]
Thanks for the review @reta, I addressed the comments. I closed the previous PR (#981) too.

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
Thank you, @awrb !

**Comment by reta** [comment]
Hm ... the change introduced flakyness into `osgi/itests` [1], [2] [3], also seeing it flipping locally, looking into it right now:

[1] https://ci-builds.apache.org/job/CXF/job/pipeline/job/3.5.x-fixes/270/
[2] https://ci-builds.apache.org/job/CXF/job/pipeline/job/3.5.x-fixes/269/
[3] https://ci-builds.apache.org/job/CXF/job/pipeline/job/3.5.x-fixes/267/

**Comment by awrb** [comment]
Oops.. it never failed for me locally before. I have run the tests in `osgi/itests` several times now (via `mvn clean test` or `install`), but it always passes, so I cannot reproduce this.

The pipeline reports
`No message body writer has been found for class org.apache.cxf.osgi.itests.jaxrs.Book`
and indeed there is no XML provider in set up for JAXRSServerFactoryBean. Should XML work OOTB without a provider? If you were using JSON, you would have to add JacksonJsonProvider, so I would expect the same for XML.

Can you look at #983 to see if it helps?

**Comment by reta** [comment]
Thanks @awrb, it seems to be a client-side issue (not the server side).

> Should XML work OOTB without a provider? 

Yes, JAXB should be there

**Pull request merged**